[BUG]  geneweb do not build with ocaml 4.08 since merge of PR #1751

geneweb do not build with ocaml 4.08 since merge of PR #1751

* As per dune-project, we are supposed to be able to build with `(ocaml (>= 4.08))`
* BUT if I try to build with ocaml 4.08.1, then build failed with `Unbound value Unix.realpath`
* git blame is pointing to cid 4725935bc3, identified as part of specified PR.
* note that this problem was already reported by @a2line  as comment of this PR with no answer.
```
Generating bin/gwrepl/.depend…File "lib/gwdb-legacy/database.ml", line 950, characters 17-30:
950 |     let bfname = Unix.realpath bname in
                       ^^^^^^^^^^^^^
Error: Unbound value Unix.realpath
make: *** [Makefile:45 : bin/gwrepl/.depend] Erreur 1
```
```
$git blame lib/gwdb-legacy/database.ml |grep 950
4725935bc3 lib/gwdb-legacy/database.ml (Keryan Didier   2024-03-22 11:17:54 +0100  950)     let bfname = Unix.realpath bname in
```

--- a/lib/gwdb-legacy/database.ml
+++ b/lib/gwdb-legacy/database.ml
@@ -947,9 +947,12 @@ let opendb ?(read_only = false) bname =
         im_descends,
         im_strings )
         : ro_data_records =
-    let bfname = Unix.realpath bname in
+    let bid =
+      let s = Unix.stat bname in
+      s.st_dev, s.st_ino
+    in
     match
-      List.find_opt (fun (n, _) -> String.equal bfname n) !cached_records
+      List.find_opt (fun (n, _) -> bid = n) !cached_records
     with
     | Some (_, records) -> records
     | None ->
@@ -1004,7 +1007,7 @@ let opendb ?(read_only = false) bname =
         let dr =
           (persons, ascends, unions, families, couples, descends, strings)
         in
-        if read_only then cached_records := (bfname, dr) :: !cached_records;
+        if read_only then cached_records := (bid, dr) :: !cached_records;
         dr
   in
   let persons =

--- a/lib/gwdb-legacy/database.ml
+++ b/lib/gwdb-legacy/database.ml
@@ -949,11 +949,9 @@ let opendb ?(read_only = false) bname =
         : ro_data_records =
     let bid =
       let s = Unix.stat bname in
-      s.st_dev, s.st_ino
+      (s.st_dev, s.st_ino)
     in
-    match
-      List.find_opt (fun (n, _) -> bid = n) !cached_records
-    with
+    match List.find_opt (fun (n, _) -> bid = n) !cached_records with
     | Some (_, records) -> records
     | None ->
         let persons =

